contracts/Certificate.sol

```diff
@@ -133,8 +163,13 @@ contract Certificate is
   )
     internal
     override(ERC1155PresetMinterPauserUpgradeable, ERC1155SupplyUpgradeable)
+    whenNotPaused
   {
-    if (from != address(0)) {
+    bool isMinting = from != address(0);
+    bool isBurning = to != address(0);
+    if (
+      (isMinting || isBurning) || !hasRole(CERTIFICATE_OPERATOR_ROLE, operator)
```

Comment:
This doesn't entirely make sense either.  Looks like a late refactor impacted this?

If the flags were implemented as named then this would be:

`!(isMinting || isBurning || hasRole(CERTIFICATE_OPERATOR_ROLE, operator))`